src/input/source/led/multicolor_chassis.rs

```diff
@@ -0,0 +1,167 @@
+            .map_err(|err| Box::new(MultiColorChassisError::MultiIndexError(err)))?;
+        let multi_index_strings = contents
+            .split_whitespace()
+            .into_iter()
+            .map(|str| String::from(str))
+            .collect::<Vec<String>>();
+        let mut multi_index_map = Vec::<IndexType>::with_capacity(multi_index_strings.len());
+        for idx in multi_index_strings.iter() {
+            match idx.as_str() {
+                "rgb" => multi_index_map.push(IndexType::RGB),
+                _ => {
+                    return Err(Box::new(MultiColorChassisError::UnsupportedIndexType(
+                        idx.clone(),
+                    )))
+                }
+            }
+        }
+
+        let result = Self {
+            fixed_color,
+            multi_intensity_path,
+            multi_index_map,
+        };
+
+        // Immediatly set the user-defined color if one is provided
+        if let Some(color) = result.fixed_color.as_ref() {
+            result.write_color(color.r, color.g, color.b)?
+        };
+
+        Ok(result)
+    }
+
+    fn write_color(&self, r: u8, g: u8, b: u8) -> Result<(), Box<dyn Error + Send + Sync>> {
+        let contents = std::fs::read_to_string(self.multi_intensity_path.as_path())
+            .map_err(|err| Box::new(MultiColorChassisError::MultiIndexError(err)))?;
+        let contents = (self
+            .multi_index_map
+            .iter()
+            .zip(contents.split_whitespace().into_iter())
+            .map(|(index_type, index_value)| match index_type {
+                IndexType::RGB => {
+                    (((r as u32) << 16u32) | ((g as u32) << 8u32) | ((b as u32) << 0u32))
+                        .to_string()
+                }
+                _ => String::from(index_value),
+            })
+            .collect::<Vec<String>>())
+        .join(" ");
+        Ok(
+            std::fs::write(self.multi_intensity_path.as_path(), contents)
+                .map_err(|err| Box::new(MultiColorChassisError::MultiIntensityUpdateError(err)))?,
+        )
+    }
+}
+
+impl Debug for MultiColorChassis {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MultiColorChassis").finish()
+    }
+}
+
+impl SourceInputDevice for MultiColorChassis {
+    fn poll(
+        &mut self,
+    ) -> Result<Vec<crate::input::event::native::NativeEvent>, crate::input::source::InputError>
+    {
+        Ok(Vec::new())
+    }
+
+    fn get_capabilities(
+        &self,
+    ) -> Result<Vec<crate::input::capability::Capability>, crate::input::source::InputError> {
+        Ok(Vec::new())
+    }
+}
+
+impl SourceOutputDevice for MultiColorChassis {
+    fn write_event(
+        &mut self,
+        event: crate::input::output_event::OutputEvent,
+    ) -> Result<(), crate::input::source::OutputError> {
+        //log::trace!("Received output event: {event:?}");
+        let _ = event;
+        Ok(())
+    }
```

Comment:
We should probably handle output events so LEDs can be controlled. Or we can handle this in a follow-up PR.

```suggestion
    fn write_event(&mut self, event: OutputEvent) -> Result<(), crate::input::source::OutputError> {
        //log::trace!("Received output event: {event:?}");
        match event {
            OutputEvent::DualSense(report) => {
                if !report.allow_led_color {
                    return Ok(());
                }
                // Handle DualSense LED events...
            }
            _ => {
                return Ok(());
            }
        }

        Ok(())
    }
```

Reply:
I was unable to figure out why so I left it alone for this PR, but I will look into getting it in a follow-up commit in this PR